packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js

```diff
@@ -186,11 +186,20 @@ const reducer = (state, action) =>
         const newRelations = [...modifiedDataRelations];
 
         if (action.type === 'REORDER_RELATION') {
-          const [newKey] = generateNKeysBetween(
-            modifiedDataRelations[newIndex - 1]?.__temp_key__,
-            modifiedDataRelations[newIndex]?.__temp_key__,
-            1
-          );
+          // Swap the order of the arguments to ensure the lesser value is always passed first to generateNKeysBetween
+          const newKey = (
+            oldIndex > newIndex
+              ? generateNKeysBetween(
+                  modifiedDataRelations[newIndex - 1]?.__temp_key__,
+                  modifiedDataRelations[newIndex]?.__temp_key__,
+                  1
+                )
+              : generateNKeysBetween(
+                  modifiedDataRelations[newIndex]?.__temp_key__,
+                  modifiedDataRelations[newIndex + 1]?.__temp_key__,
+                  1
+                )
+          )[0];
```

Comment:
```suggestion
					const startKey = oldIndex > newIndex ? modifiedDataRelations[newIndex - 1]?.__temp_key__ : modifiedDataRelations[newIndex]?.__temp_key__;
					
					const endKey = oldIndex > newIndex ? modifiedDataRelations[newIndex]?.__temp_key__ : modifiedDataRelations[newIndex + 1]?.__temp_key__;
					
					const [newKey] = generateNKeysBetween(
                  startKey,
                  endKey,
                  1
                )
```

I think this reads better imo even though it's a bit more verbose it's clearer what we're trying to do. If you commit the comment i suspect you'll need to format the code so the CI passes.

wdyt?

Reply:
You need to update the lock file by running `yarn`. Also can you rebase your PR off main please, I can't make any changes to your PR for some reason.

Reply:
@joshuaellis  Thanks. but It happens the same on main branch

Reply:
I've just pushed to `main` can you rebase please?

Reply:
@joshuaellis Thank for your help.